azure-pipelines.yml

```diff
@@ -485,136 +301,39 @@ stages:
-        SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.5-bin-hadoop2.7
-
-    - task: DotNetCoreCLI@2
-      displayName: 'E2E tests for Spark 2.4.6'
-      inputs:
-        command: test
-        projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
-        arguments: '--configuration $(buildConfiguration) --filter $(backwardCompatibleTestsToFilterOut)'
-      env:
-        SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.6-bin-hadoop2.7
-
-    - task: DotNetCoreCLI@2
-      displayName: 'E2E tests for Spark 2.4.7'
-      inputs:
-        command: test
-        projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
-        arguments: '--configuration $(buildConfiguration) --filter $(TestsToFilterOut)'
-      env:
-        SPARK_HOME: $(Build.BinariesDirectory)\spark-2.4.7-bin-hadoop2.7
+    
+    - template: azure-pipelines-tests-template.yml
+      parameters:
+        versions: 
+        - '2.3.0'
+        - '2.3.1'
+        - '2.3.2'
+        - '2.3.3'
+        - '2.3.4'
+        - '2.4.0'
+        - '2.4.1'
+        - '2.4.3'
+        - '2.4.4'
+        - '2.4.5'
+        - '2.4.6'
+        - '2.4.7'
+        testOptions: '--filter $(backwardCompatibleTestsToFilterOut)'
 
     # Spark 3.0.* uses Arrow 0.15.1, which contains a new Arrow spec. This breaks backward
     # compatibility when using Microsoft.Spark.Worker with incompatible versions of Arrow.
     # Skip Arrow tests until the backward compatibility Worker version is updated.
-    - task: DotNetCoreCLI@2
-      displayName: 'E2E tests for Spark 3.0.0'
-      inputs:
-        command: test
-        projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
-        arguments: "--configuration $(buildConfiguration) --filter $(backwardCompatibleTestsToFilterOut)&\
+    - template: azure-pipelines-tests-template.yml
+      parameters:
+        versions:
+        - '3.0.0'
+        testOptions: "--filter $(backwardCompatibleTestsToFilterOut)&\
         (FullyQualifiedName!=Microsoft.Spark.E2ETest.IpcTests.DataFrameTests.TestGroupedMapUdf)&\
         (FullyQualifiedName!=Microsoft.Spark.E2ETest.IpcTests.DataFrameTests.TestVectorUdf)"
-      env:
-        SPARK_HOME: $(Build.BinariesDirectory)\spark-3.0.0-bin-hadoop2.7
 
-    - task: DotNetCoreCLI@2
-      displayName: 'E2E tests for Spark 3.0.1'
-      inputs:
-        command: test
-        projects: '**/Microsoft.Spark*.E2ETest/*.csproj'
-        arguments: "--configuration $(buildConfiguration) --filter $(TestsToFilterOut)&\
+    - template: azure-pipelines-tests-template.yml
+      parameters:
+        versions:
+        - '3.0.1'
+        testOptions: "--filter $(TestsToFilterOut)&\
         (FullyQualifiedName!=Microsoft.Spark.E2ETest.IpcTests.DataFrameTests.TestGroupedMapUdf)&\
-        (FullyQualifiedName!=Microsoft.Spark.E2ETest.IpcTests.DataFrameTests.TestVectorUdf)"
-      env:
-        SPARK_HOME: $(Build.BinariesDirectory)\spark-3.0.1-bin-hadoop2.7
+        (FullyQualifiedName!=Microsoft.Spark.E2ETest.IpcTests.DataFrameTests.TestVectorUdf)"
```

Comment:
I think we can combine these.  The `3.0.1` case should actually use `backwardCompatibleTestsToFilterOut` instead of `TestsToFilterOut`

Reply:
Ok @suhsteve I've made that change